edisgo/tools/temporal_complexity_reduction.py

```diff
@@ -42,31 +51,48 @@ def _scored_most_critical_loading(edisgo_obj: EDisGo) -> pd.Series:
     # Get lines that have violations
     crit_lines_score = relative_i_res[relative_i_res > 1]
 
-    # Get most critical timesteps per component
+    # Get most critical time steps per component
+    crit_lines_score = crit_lines_score[crit_lines_score == crit_lines_score.max()]
+
+    if weight_by_costs:
+        # weight line violations with expansion costs
+        costs = _costs_per_line_and_transformer(edisgo_obj)
+        crit_lines_score = crit_lines_score * costs.loc[crit_lines_score.columns]
```

Comment:
That's a good point. I thought about for a while and I think both the way we do it now and what you are suggesting are not really correct as the relative loading doesn't convey the number of additional lines needed to solve the problem. 
E.g. when there are two 1 MW lines and the loading is 3.2 MW, then we would need two more lines of the same type to solve the problem. So in that case subtracting 1 and ceiling would not be correct, because relative loading is 1.6 and subtracting 1 and ceiling would only yield 1 line. But when there is only one 1 MW line and the loading is 3.2 MW, then subtracting 1 and ceiling would yield the right number of lines.
The correct way would probably be to use the absolute overloading and the s_nom of the standard line in the respective voltage level. But I think since this should just be an estimation of the costs I would leave it like it is for now. I will make a ToDo there, so we can maybe change it at some point.

Reply:
Ah true, I did not think of parallel lines. Yes I guess it is fine for now and we can think about adapting it some other time.